boot/zephyr/nrf_nv_counters.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <stdint.h>
+#include <bl_otp_counters.h>
+#include "bootutil/fault_injection_hardening.h"
+
+fih_int boot_nv_security_counter_init(void){
+    return FIH_SUCCESS;
+}
+
+/**
+ * Reads the stored value of a given image's security counter.
+ *
+ * @param image_id          Index of the image (from 0).
+ * @param security_cnt      Pointer to store the security counter value.
+ *
+ * @return                  FIH_SUCCESS on success
+ */
+fih_int boot_nv_security_counter_get(uint32_t image_id, fih_int *security_cnt){
+    int err;
+    uint16_t cur_sec_cnt;
+
+    if(security_cnt == NULL){
+        return FIH_FAILURE;
+    }
+
+    switch (image_id)
+    {
+    case 0:
+        err = get_monotonic_counter(COUNTER_DESC_MCUBOOT_HW_COUNTER_ID0, &cur_sec_cnt);
+        break;
+    case 1:
+        err = get_monotonic_counter(COUNTER_DESC_MCUBOOT_HW_COUNTER_ID1, &cur_sec_cnt);
+        break;
+    case 2:
+        err = get_monotonic_counter(COUNTER_DESC_MCUBOOT_HW_COUNTER_ID2, &cur_sec_cnt);
+        break;
+    default:
+        return FIH_FAILURE;
+    }
+
```

Comment:
```suggestion
   if(image_id > 2) {
     return FIH_FAILURE;
   }
   
   uint32_t? ids[3] = {
     COUNTER_DESC_MCUBOOT_HW_COUNTER_ID0,
     COUNTER_DESC_MCUBOOT_HW_COUNTER_ID1,
     COUNTER_DESC_MCUBOOT_HW_COUNTER_ID2
  };
               
   err = get_monotonic_counter(ids[image_id], &cur_sec_cnt);
```

For readability and maintainability.

Reply:
I see we can do the same refactoring below.

So ids can be put into file-level scope (with static const to ensure it's placed in flash).

Reply:
That makes sense, I can do this.